### PR TITLE
Add Exercise04

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

File: Exercise04/Program.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Exercise04
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Func<string, Func<string, bool>, List<string>> FilterFile = (path, endWith) =>
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: Just need to make a function.

Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

File: Exercise04/Program.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        static void Main(string[] args)
+        {
+            var pathFolder = @"C:\Users\huynh\OneDrive";
+            var newList = FilterFile(pathFolder, x =>x.EndsWith(".txt"));
+            newList.ForEach(Console.WriteLine);
+
+            Console.ReadKey();
+        }
+
+        public static List<string> FilterFile(string path, Func<string, bool> endWith)
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: `Func<string, bool>` ==> `Predicate<string>`

File: Exercise04/Program.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            var files = Directory.GetFiles(path, "*", SearchOption.AllDirectories);
+
+            var list = new List<string>();
+
+            foreach (var i in files)
```
Comment (voquanghoa) [inline Exercise04/Program.cs]: `i` ==> `file`

File: Exercise04/Program.cs (hunk 2 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

File: Exercise04/Program.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise04/Program.cs]: Done

**Pull request merged**